### PR TITLE
fix: `useAsync` dependency arrays

### LINKED ISSUE #2061: Add exhaustive dependency check for `useAsync` hook
## What is this about

We don't have any eslint rules in place to check dependencies when the `useAsync` hook is being used. This can be fixed by extending the eslint rules with:

```
"react-hooks/exhaustive-deps": ["warn", {
    "additionalHooks": "useAsync"
}]
```

Extending this rule will result in a few warnings that need to be addressed.

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #2061

## How this PR fixes it

This adds the exhaustive dependency array linter to our `useAsync` hook. All relative findings have been adjusted accordingly.

Dependency arrays that needed changing were found in:

1. `DecodedTx`
2. `useRedefine`
3. `useGasPrice`
4. `useIsValidExecution`
5. `useWalletCanRelay`

## How to test it

According to the above list:

1. Observe the transaction info displaying in the sign/execute form (above the balance change).
2. Redefine works.
3. Gas estimation works in Safe creation.
4. Invalid executions are warned about in the execution form ("This transaction will most likely fail...").
5. It is _not_ possible to relay 6 times in an hour.

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

**Review by schmanu — APPROVED** [review]
A very necessary change!
Good work :)

This could have some side effects though, which may be hard to find. Our current release is already quite big. I'd therefore want to discuss if we should include it in the following release. WDYT?

**Comment by iamacook** [comment]
> This could have some side effects though, which may be hard to find. Our current release is already quite big. I'd therefore want to discuss if we should include it in the following release. WDYT?

I only adjusted the dependencies of five hooks which are listed in the description but I totally agree. Let's leave it until the next release and see if QA find any issues until then.

File: src/hooks/loadables/useLoadSpendingLimits.ts [hunk 1]
```diff
@@ -105,7 +105,7 @@ export const useLoadSpendingLimits = (): AsyncResult<SpendingLimitState[]> => {
     if (!provider || !safeLoaded || !safe.modules || !tokenInfoFromBalances) return
 
     return getSpendingLimits(provider, safe.modules, safeAddress, chainId, tokenInfoFromBalances)
-  }, [provider, safeLoaded, safe.modules?.length, safeAddress, chainId, safe.txHistoryTag, tokenInfoFromBalances])
```
Comment (usame-algan) [inline src/hooks/loadables/useLoadSpendingLimits.ts:108]: Are we sure the `txHistoryTag` needs to be removed here? I remember we wanted to automatically update the value on the settings page without reloading

**Comment by francovenica** [comment]
LGTM 

Tried few tx checking the details
Created several safes, the initial estimation of gas worked fine every time
Redefine works fine
Trying to execute tx that will fail shows the warning properly
Relaying only works 5 times per hour

File: src/hooks/loadables/useLoadSpendingLimits.ts (hunk 1 above)
Comment (iamacook) [reply src/hooks/loadables/useLoadSpendingLimits.ts:108]: True, I understand the requirement. I've added it back.

**Review by usame-algan — APPROVED** [review]

**Pull request merged**